lib/agent/reporter.rb

```diff
@@ -33,28 +46,44 @@ def push(data, non_block = true)
         end
       end
 
+      # Closes down reporter
+      # Attempts to shutdown gracefully, but will force close if it takes too long
+      def close
+        self.class.instance_variable_get(:@singleton__mutex__).synchronize do
+          @report_queue.close
```

Comment:
`SizedQueue#close` only exists in Ruby 2.3.0 and newer, need to figure out how to close the queue in earlier rubies.